src/main/java/ai/elimu/web/SignOnControllerDiscord.java

```diff
@@ -132,10 +132,20 @@ public String handleCallback(HttpServletRequest request) throws IOException, Int
                 String id = String.valueOf(idAsLong);
                 contributor.setProviderIdDiscord(id);
             }
+
             if (jsonObject.has("avatar")) {
-                String uriAvatar = "https://cdn.discordapp.com/avatars/" + jsonObject.getLong("id") + "/" + jsonObject.getString("avatar") + ".png";
-                logger.info(uriAvatar);
-                contributor.setImageUrl(uriAvatar);
+                if (jsonObject.get("avatar") instanceof String) {
+                    String avatar = jsonObject.getString("avatar");
+                    if (!avatar.isEmpty()) {
+                        String uriAvatar = "https://cdn.discordapp.com/avatars/" + jsonObject.getLong("id") + "/" + avatar + ".png";
+                        logger.info(uriAvatar);
+                        contributor.setImageUrl(uriAvatar);
+                    } else {
+                      logger.info("Avatar field is empty.");
+                    } 
+                } else {
+                    logger.warn("Avatar field is not a string: " + jsonObject.get("avatar"));
```

Comment:
@venkatesh2k3 What does the JSON response look like when this scenario ("Avatar field is not a string") occurs?